app/components/close_help_post.py

```diff
@@ -123,15 +128,73 @@ async def close_post(
         return
 
     await interaction.response.defer(ephemeral=True)
+    followup = "Post closed."
 
     desired_tag_id = config.HELP_CHANNEL_TAG_IDS[tag]
     await post.add_tags(next(tag for tag in help_tags if tag.id == desired_tag_id))
 
     if title_prefix is None:
         title_prefix = f"[{tag.upper()}]"
-    await post.edit(name=f"{title_prefix} {post.name}")
+
+    try:
+        await post.edit(name=f"{title_prefix} {post.name}")
+    except discord.HTTPException as e:
+        # Re-raise if it's not because the new post title was invalid.
+        if e.status != INVALID_REQUEST_DATA or e.code != INVALID_FORM_BODY:
+            raise
+
+        # HACK: there does not appear to be any way to get the actual error
+        # without parsing the returned string or using a private field.  This
+        # is likely a limitation of discord.py, as the Discord API
+        # documentation mentions:
+        #     Some of these errors may include additional details in the form
+        #     of Error Messages provided by an errors object.
+        # in https://discord.com/developers/docs/topics/opcodes-and-status-codes#json
+        # Both approaches are going to be tried... here be dragons.
+        try:
+            # The format of e._errors that is being parsed below is
+            # {
+            #     "name": {
+            #         "_errors": [
+            #             {
+            #                 "code": "BASE_TYPE_MAX_LENGTH",
+            #                 "message": "Must be 100 or fewer in length.",
+            #             }
+            #         ]
+            #     }
+            # }
+            # This tries to make as few assumptions of the keys of the
+            # dictionary as possible, as the field is private.
+            returned_error = next(iter(next(iter(e._errors.values())).values()))[0][  # noqa: SLF001
+                "message"
+            ]
+        except (StopIteration, IndexError):
+            try:
+                # Parsing something of the form:
+                # "Invalid Form Body\nIn name: Must be 100 or fewer in length."
+                returned_error = (
+                    str(e).splitlines()[1].removeprefix("In name: ").strip()
+                )
+            except IndexError:
+                returned_error = None
```

Comment:
This can be much cleaner with pattern matching:
```suggestion
        match e._errors, str(e).splitlines():
            case {"name": {"_errors": [{"code": _, "message": msg}]}}, _:
                returned_error = msg
            case _, [_, snd]:
                # Parsing something of the form:
                # "Invalid Form Body\nIn name: Must be 100 or fewer in length."
                returned_error = snd.removeprefix("In name: ").strip()
            case _:
                returned_error = None
```

Reply:
Hmm, I just realised that pattern matching results in the exact structure of the private field being hardcoded in the code too, while my original thing tried to assume as little as possible about the private field.  If we're going to go this route, I may as well just directly access the element itself.

Reply:
Well that cleaned up the code a whole bunch… it now looks completely different from the original PR.

Reply:
The thing with direct access is that it's not safeguarded. E.g. if the name `_errors` changes, this code simply won't match and will try the next case, while direct access will crash the entire thing.

Reply:
I brought back the `except IndexError` though.

Reply:
Actually I had handled the wrong errors there, fixed now.

Reply:
My fix for the wrong errors was wrong too, fixed yet again.

Reply:
Though it could be argued that my list of exceptions isn't exhaustive, and pattern matching would be the Ruff-friendly™ way to say `except Exception: pass`…